src/rte/mo_rte_sw.jl

```diff
@@ -28,105 +28,131 @@
       error_msg = "rte_sw: inc_flux inconsistently sized"
-    if(any_vals_less_than(inc_flux, 0._wp)) &
+    end
+
+    if(any_vals_less_than(inc_flux, wp(0))) 
       error_msg = "rte_sw: one or more inc_flux < 0"
-    if(present(inc_flux_dif)) then
-      if(any([size(inc_flux_dif, 1),    size(inc_flux_dif, 2)]    /= [ncol, ngpt])) &
+    end
+
+    if(present(inc_flux_dif))
+      if(any([size(inc_flux_dif)[1], size(inc_flux_dif)[2]] != [ncol, ngpt]))
         error_msg = "rte_sw: inc_flux_dif inconsistently sized"
-      if(any_vals_less_than(inc_flux_dif, 0._wp)) &
+      end
+      if(any_vals_less_than(inc_flux_dif, wp(0)))
         error_msg = "rte_sw: one or more inc_flux_dif < 0"
-    end if
+      end
+    end
 
-    if(any([size(sfc_alb_dir, 1), size(sfc_alb_dir, 2)] /= [nband, ncol])) &
+    if(any([size(sfc_alb_dir)[1], size(sfc_alb_dir)[2]] != [nband, ncol])) 
       error_msg = "rte_sw: sfc_alb_dir inconsistently sized"
-    if(any_vals_outside(sfc_alb_dir,  0._wp, 1._wp)) &
+    end
+    if(any_vals_outside(sfc_alb_dir,  wp(0), wp(1)))
       error_msg = "rte_sw: sfc_alb_dir out of bounds [0,1]"
-    if(any([size(sfc_alb_dif, 1), size(sfc_alb_dif, 2)] /= [nband, ncol])) &
+    end
+    if(any([size(sfc_alb_dif)[1], size(sfc_alb_dif)[2]] != [nband, ncol])) 
       error_msg = "rte_sw: sfc_alb_dif inconsistently sized"
-    if(any_vals_outside(sfc_alb_dif,  0._wp, 1._wp)) &
+    end
+    if(any_vals_outside(sfc_alb_dif, wp(0), wp(1)))
       error_msg = "rte_sw: sfc_alb_dif out of bounds [0,1]"
+    end
 
-    if(len_trim(error_msg) > 0) then
-      if(len_trim(atmos%get_name()) > 0) &
-        error_msg = trim(atmos%get_name()) // ': ' // trim(error_msg)
+    if(length( strip(error_msg) ) > 0)
+      if(length(strip(get_name(atmos))) > 0)
+        error_msg = strip(get_name(atmos)) * ": " * strip(error_msg)
+      end
       return
-    end if
-
+    end
     # ------------------------------------------------------------------------------------
-    allocate(gpt_flux_up (ncol, nlay+1, ngpt), gpt_flux_dn(ncol, nlay+1, ngpt), gpt_flux_dir(ncol, nlay+1, ngpt))
-    allocate(sfc_alb_dir_gpt(ncol, ngpt), sfc_alb_dif_gpt(ncol, ngpt))
+    gpt_flux_up  = Array{wp}(undef,ncol, nlay+1, ngpt)
+    gpt_flux_dn  = Array{wp}(undef,ncol, nlay+1, ngpt)
+    gpt_flux_dir = Array{wp}(undef,ncol, nlay+1, ngpt)
+
+    sfc_alb_dir_gpt = Array{wp}(undef,ncol, ngpt)
+    sfc_alb_dif_gpt = Array{wp}(undef,ncol, ngpt)
     # ------------------------------------------------------------------------------------
     # Lower boundary condition -- expand surface albedos by band to gpoints
     #   and switch dimension ordering
-
-    #$acc enter data create(sfc_alb_dir_gpt, sfc_alb_dif_gpt)
-    call expand_and_transpose(atmos, sfc_alb_dir, sfc_alb_dir_gpt)
-    call expand_and_transpose(atmos, sfc_alb_dif, sfc_alb_dif_gpt)
+#    #$acc enter data create(sfc_alb_dir_gpt, sfc_alb_dif_gpt)
+    expand_and_transpose(atmos, sfc_alb_dir, sfc_alb_dir_gpt)
```

Comment:
```suggestion
    expand_and_transpose!(atmos, sfc_alb_dir, sfc_alb_dir_gpt)
```
Since this function call is changing variables in-place, we know it needs a `!` at the end of the function name